From c80318ae05cd03b85f893d4ca6567c39741991a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=9D=A8=E6=96=87=E6=98=8E?= <yangwenming@bytedance.com>
Date: Sun, 30 Jul 2023 16:20:29 +0800
Subject: [PATCH] [riscv][tagged-ptr] Change obj->IsFoo() to IsFoo(obj)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Port commit 9c869941764bc57a9a3b86a090323236d9b3af7c

Removes the IsFoo predicates from Object, HeapObject, and Tagged<T>, and
replaces them with free functions that are passed a Tagged<T>.

This allows a coding style which doesn't require dereferencing of
Objects that could be Smis.

Bug: v8:12710
Change-Id: I1d7473f00147470e08865732db65daba4c86e19e
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4727049
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Auto-Submit: 杨文明 <yangwenming@bytedance.com>
Reviewed-by: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#89264}
---
 src/execution/riscv/simulator-riscv.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/execution/riscv/simulator-riscv.cc b/src/execution/riscv/simulator-riscv.cc
index 56ee75bfda2..0eb9cbaf98e 100644
--- a/src/execution/riscv/simulator-riscv.cc
+++ b/src/execution/riscv/simulator-riscv.cc
@@ -1940,10 +1940,10 @@ void RiscvDebugger::Debug() {
                  reinterpret_cast<intptr_t>(cur), *cur, *cur);
           Object obj(*cur);
           Heap* current_heap = sim_->isolate_->heap();
-          if (obj.IsSmi() ||
+          if (IsSmi(obj) ||
               IsValidHeapObject(current_heap, HeapObject::cast(obj))) {
             PrintF(" (");
-            if (obj.IsSmi()) {
+            if (IsSmi(obj)) {
               PrintF("smi %d", Smi::ToInt(obj));
             } else {
               obj.ShortPrint();
-- 
2.35.1

